charts/gateway-helm/templates/envoy-gateway-hpa.yaml

```diff
@@ -0,0 +1,37 @@
+{{- if .Values.hpa.enabled }}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: envoy-gateway
+  namespace: '{{ $.Release.Namespace }}'
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: envoy-gateway
+  {{- if .Values.hpa.minReplicas }}
+  minReplicas: {{ .Values.hpa.minReplicas }}
+  {{- end }}
+  maxReplicas: {{ .Values.hpa.maxReplicas }}
+  {{- if .Values.hpa.behavior }}
+  behavior:
+{{ toYaml .Values.hpa.behavior | indent 4 }}
+  {{- end }}
+  metrics:
+    {{- if .Values.hpa.targetCPU }}
```

Comment:
Prefer to keep this entirely in values (similar to behavior) and not force end users to rely only on CPU/Mem. For example, custom metrics could be relevant here, e.g. size of the XDS cache, number of connected envoy proxies, etc.